lib/identity/client.js

```diff
@@ -141,6 +139,35 @@ class Client extends PartialClient {
     const resultQuery = await this.storage.search(request)
     return resultQuery
   }
+  async updateSecret(secret) {
+    // Search for all records of the same type
+    const request = new Search(true)
+    let searchType = `tozny.secret.${SECRET_UUID}.${secret.secretType}.${secret.secretName}`
+    request.match({ type: searchType })
+    const resultQuery = await this.storage.search(request)
```

Comment:
1.) you shouldn’t have to do a search to find the previous latest version of this secret because the user is passing in the secret to update
2.) If you do need to find that secret from a search, talk to @willmichael about how to construct this search to be more focused (e.g. searching for records not only of the record type but with a tag of version:latest)